res/res_pjsip_refer.c

```diff
@@ -733,6 +736,49 @@ static void refer_blind_callback(struct ast_channel *chan, struct transfer_chann
 
 	pbx_builtin_setvar_helper(chan, "SIPREFERRINGCONTEXT", S_OR(refer->context, NULL));
 
+	get_xfrdata = pbx_builtin_getvar_helper(chan, "GET_TRANSFERRER_DATA");
+	if (!ast_strlen_zero(get_xfrdata)) {
+		const pjsip_msg * msg = refer->rdata->msg_info.msg;
+		const struct pjsip_hdr *end = &msg->hdr;
+		struct pjsip_hdr *hdr = end->next;
+		struct ast_str *pbxvar = ast_str_alloca(MAX_HDR_SIZE + 1);
+		char *hdr_name = ast_alloca(MAX_HDR_SIZE + 1);
+		int hdr_name_len;
+		const char *prefix = get_xfrdata;
+		/* The '*' alone matches all headers. */
+		if (strcmp(prefix, "*") == 0) {
+			prefix = "";
+		}
+		
+		for (; hdr!=end; hdr = hdr->next) {
+			hdr_name_len = MIN(pj_strlen(&hdr->name), MAX_HDR_SIZE);
+			ast_copy_pj_str(hdr_name, &hdr->name, MAX_HDR_SIZE + 1);
+
+			if (ast_begins_with(hdr_name, prefix) == 1) {
```

Comment:
Could pj_strnicmp2 be used instead so that on every iteration the string doesn't need to be copied again, and thus only copied when actually needed?